### DISCUSSION TITLE
Solis- No devices or entities- Newbie

### DISCUSSION THREAD

**Discussion by Richard7619** [wills106/homeassistant-solax-modbus#450] [Q&A] [answered]
Hi I'm a Newbie to HA but with help on here I’m making progress. 

I have installed the hardware and worked through this project
[https://github.com/Jumpy07/Solis---SolisCloud-and-Home-Assistant ](url)

When I get to the Solax modbus integration I can not get devices or entities to load. 

My Inverter is a Solis RHI-48ES-5G-DC, serial number starts 6031. 
I checked the solis plugin in file editor and that serial number is listed. 

I switched on the debug log and get the following:

  File "/usr/local/lib/python3.11/site-packages/pymodbus/client/base.py", line 191, in execute
    raise ConnectionException(f"Failed to connect[{str(self)}]")
pymodbus.exceptions.ConnectionException: Modbus Error: [Connection] Failed to connect[ModbusTcpClient(192.168.1.218:502)]
2023-06-20 18:42:04.445 WARNING (MainThread) [custom_components.solax_modbus.plugin_solis] Solis: reading serial number from address 0x80ec failed; other address may succeed
2023-06-20 18:42:04.448 ERROR (MainThread) [custom_components.solax_modbus.plugin_solis] Solis: cannot find serial number, even not for other Inverter
2023-06-20 18:42:04.450 ERROR (MainThread) [custom_components.solax_modbus.plugin_solis] unrecognized Solis inverter type - serial number : unknown
2023-06-20 18:51:05.705 ERROR (MainThread) [pymodbus.logging] Connection to (192.168.1.218, 502) failed: [Errno 111] Connection refused
2023-06-20 18:51:05.708 WARNING (MainThread) [custom_components.solax_modbus.plugin_solis] Solis: attempt to read serialnumber failed at 0x80ec
Traceback (most recent call last):
  File "/config/custom_components/solax_modbus/plugin_solis.py", line 53, in _read_serialnr
    inverter_data = hub.read_input_registers(unit=hub._modbus_addr, address=address, count=8)

any help would be appreciated.

**Comment by alienatedsec** [comment]
@Richard7619 there is no connection between HA and your Modbus interface. It's either port, protocol or IP wrongly configured.

**Reply by Richard7619** [reply]
This is the status of the Elfin EW11. Should sent and received frames be on different? 
![DA31734E-86AA-41F3-B300-89FF55B1C5FB](https://github.com/wills106/homeassistant-solax-modbus/assets/135166611/a8cf9eb0-cc27-4eeb-84c9-365613cc16d9)
![FD66DE42-4CF6-4EA1-8E28-E5DD39C3E64C](https://github.com/wills106/homeassistant-solax-modbus/assets/135166611/2196d76a-417b-46eb-b5ba-45281b683dea)

**Reply by alienatedsec** [reply]
@Richard7619 It's hard to advise as I don't use that solution. It would be best if you asked for connectivity and configuration help at the project's source that you have followed.

**Reply by Richard7619** [reply, marked as answer]
It turned out to be a wiring issue. I now see 1 device and 143 sensors.

**Reply by Richard7619** [reply]
@alienatedsec  The Solax integration now shows list of controls, sensors, configuration and diagnostic but the status of all sensors is "unknown". The solis cloud is pulling all the data into the solis app so any ideas on how I get the data in Solax? 
<img width="434" alt="sensors Capture" src="https://github.com/wills106/homeassistant-solax-modbus/assets/135166611/be24fd16-496e-4d95-836f-b7975c7b3675">
Thanks,

**Reply by alienatedsec** [reply]
@Richard7619 post a picture of your "2 Masters, 1 Slave hub", please. It could be that they sent you the wrong unit and it will not work as intended. Also, you can check this theory by unplugging the Datalogger and see if SolaX will work after reloading.

**Reply by Richard7619** [reply]
@alienatedsec I removed the datalogger and now I have data in Solax. I'll post photos 
<img width="271" alt="dataCapture" src="https://github.com/wills106/homeassistant-solax-modbus/assets/135166611/e201b718-0268-4a05-88c2-7f7120dc2508">

**Reply by Richard7619** [reply]
![image](https://github.com/wills106/homeassistant-solax-modbus/assets/135166611/63fda12b-0de9-4707-9d58-76c6d0d50099)

**Reply by alienatedsec** [reply]
@Richard7619 as previously said, I use something different and  @Jevburchell can share his opinion on a device you posted
